rust/loqui_client/src/connection_handler.rs

```diff
@@ -140,13 +139,19 @@ impl<E: Encoder> Handler for ConnectionHandler<E> {
         let TransportOptions { encoding, .. } = transport_options;
         // Forward Request and Push events to the connection so it can send them to the server.
         match event {
-            InternalEvent::Request { payload, waiter_tx } => {
+            InternalEvent::Request { payload, waiter } => {
                 let sequence_id = id_sequence.next();
-                self.handle_request(payload, encoding, sequence_id, waiter_tx)
+                self.handle_request(payload, encoding, sequence_id, waiter)
             }
             InternalEvent::Push { payload } => self.handle_push(payload, encoding),
         }
     }
+
+    fn handle_ping(&mut self) {
+        // Use to sweep dead waiters.
+        self.waiters
+            .retain(|_sequence_id, waiter| waiter.deadline > Instant::now());
```

Comment:
We should probably (for good measure) also notify the waiter that the request timed out.

Reply:
I tried this, but we don't have the ownership we need here in retain. I think it's fine since the waiter is most likely no longer listening or about to stop listening.